sourced/ml/mining/__main__.py

```diff
@@ -0,0 +1,56 @@
+import argparse
+import logging
+import sys
+
+from modelforge import slogging
+
+from .cmd import ArgumentDefaultsHelpFormatterNoNone
+
+
+def get_parser() -> argparse.ArgumentParser:
+    """
+    Create the cmdline argument parser.
+    """
+    parser = argparse.ArgumentParser(
+        formatter_class=ArgumentDefaultsHelpFormatterNoNone
+    )
+    parser.add_argument(
+        "--log-level",
+        default="INFO",
+        choices=logging._nameToLevel,
+        help="Logging verbosity.",
+    )
+    # Create and construct subparsers
+    subparsers = parser.add_subparsers(help="Commands", dest="command")
+
+    def add_parser(name, help_message):
+        return subparsers.add_parser(
+            name, help=help_message, formatter_class=ArgumentDefaultsHelpFormatterNoNone
+        )
+
+    return parser
+
+
+def main():
+    """
+    Create all the argparse-rs and invokes the function from set_defaults().
+
+    :return: The result of the function from set_defaults().
+    """
+    parser = get_parser()
+    args = parser.parse_args()
+    args.log_level = logging._nameToLevel[args.log_level]
+    slogging.setup(args.log_level, False)
+    try:
+        handler = args.handler
+    except AttributeError:
+
+        def print_usage(_):
+            parser.print_usage()
+
+        handler = print_usage
```

Comment:
```suggestion
        handler = lambda _: parser.print_usage()
```
Will this work? IDK, pls check

Reply:
It seems to, if not will revert later